napari/_qt/qt_viewer.py

```diff
@@ -160,8 +159,14 @@ def __init__(self, viewer: Viewer, welcome=False):
 
         # This dictionary holds the corresponding vispy visual for each layer
         self.layer_to_visual = {}
-        self.viewerButtons.consoleButton.clicked.connect(
-            self.toggle_console_visibility
+        action_manager.register_action(
+            "toggle_console_visibility",
+            self.toggle_console_visibility,
```

Comment:
How are we handling `toggles`?

Maybe `command`, should be changed to `trigger` or `toggle`and either one of them could be defined for an action.

Also how are we handling localization for the action name, if name is acting as a unique identifier?

Reply:
Also how do we handle `setIconText` for tool buttons that may provide a shorter text when displayed on a toolbar?

Reply:
we can rename `name` to `uid`, (id is keyword), and add a translatable short text.

Reply:
yes, we can likely have also something smarter for toggle, I just wanted to keep this short and improve later.

Reply:
>we can rename name to uid, (id is keyword), and add a translatable short text.

👍🏼  

>yes, we can likely have also something smarter for toggle, I just wanted to keep this short and improve later.

Maybe having `_command` would provide a clearer idea that this is in flux :) ?